pkg/apis/core/validation/validation.go

```diff
@@ -1243,13 +1243,33 @@ func validateMountPropagation(mountPropagation *core.MountPropagationMode, conta
 		return allErrs
 	}
 
-	privileged := container.SecurityContext != nil && container.SecurityContext.Privileged != nil && *container.SecurityContext.Privileged
-	if *mountPropagation == core.MountPropagationBidirectional && !privileged {
-		allErrs = append(allErrs, field.Forbidden(fldPath, "Bidirectional mount propagation is available only to privileged containers"))
+	if *mountPropagation == core.MountPropagationBidirectional {
+		// additional permissions required for mount functionality
+		if !hasBidirectionalMountPropagationPermissions(container.SecurityContext) {
+			allErrs = append(allErrs, field.Forbidden(fldPath, "Bidirectional mount propagation is available only to privileged containers and containers with the 'SYS_ADMIN' capability"))
+		}
 	}
 	return allErrs
 }
 
+func hasBidirectionalMountPropagationPermissions(securityContext *core.SecurityContext) bool {
+	if securityContext == nil {
+		// No security context configured
+		return false
+	}
+
+	// check container for required capability
+	if securityContext.Capabilities != nil {
```

Comment:
Relaxing validation has to be done over two releases. Otherwise, we allow data to be persisted that will be considered invalid by the previous release of the API server (which could still be running in a multi-server cluster during rolling upgrade).

I would suggest doing the following:

1. in `pkg/api/pod/util.go`, add an `AllowUnprivilegedBidirectionalMountPropagation bool` field to `PodValidationOptions`
2. make sure that options struct is passed through all validation functions down to this location
3. in this function, only allow the relaxed validation if the `AllowUnprivilegedBidirectionalMountPropagation` field is true
4. in `GetValidationOptionsFromPodSpecAndMeta`, compute the validation options based on the new and old object
    * in this release
      * on create (new object is non-nil, old object is nil), set the `AllowUnprivilegedBidirectionalMountPropagation` field to `false`
      * on update, set the `AllowUnprivilegedBidirectionalMountPropagation` field to `true` *only* if the old object already has bidirectional mount propagation in an unprivileged container
    * in the next release, set the `AllowUnprivilegedBidirectionalMountPropagation` field to `true` unconditionally

Reply:
Added `AllowUnprivilegedBidirectionalMountPropagation` flag and used it as described